### DISCUSSION TITLE
Can ccache caching other compiler outputs (assembly listing, *.lst files)?

### DISCUSSION THREAD

**Discussion by markrages** [ccache/ccache#1083] [Questions & answers] [answered]
I am using gcc for STM32.  Here is a snippet from a default generated STM32 project Makefile:

```makefile
 $(BUILD_DIR)/%.o: %.c Makefile | $(BUILD_DIR)
        $(CC) -c $(CFLAGS) -Wa,-a,-ad,-alms=$(BUILD_DIR)/$(notdir $(<:.c=.lst)) $< -o $@
```

This generates an assembler listing file with the "lst" extension. (as documented in "man as")

Is there a way to cache this file as well as the object file?  I am trying to avoid the situation where ccache replaces the object file, but leaves the corresponding listing file alone, so the listing refers to the wrong file.

Regards,
Mark
markrages@gmail

**Comment by jrosdahl** [comment, marked as answer]
> This generates an assembler listing file with the "lst" extension. (as documented in "man as")

That functionality was news to me. I fixed ccache to correctly consider usage of `-Wa,...=file` "too hard" (i.e. to fall back to just running the compiler) in b36a81893c6f.

> Is there a way to cache this file as well as the object file?

No, except for implementing support for this in ccache. I created #1090 for this.

**Reply by phush0** [reply]
This change made ccache unusable in almost all embedded projects using this feature. And ppl wonder why ccache is not working anymore.  I am also writing projects with STM32 and took me about 2 hours to investigate why all files are uncached. Maybe it is better to add option to turn it on or off like sloppiness

**Reply by jrosdahl** [reply]
> This change made ccache unusable in almost all embedded projects using this feature.

By "using this feature" I assume you mean they "use the compiler option but actually don't use the feature" since ccache doesn't produce any assembler listing files?

> Maybe it is better to add option to turn it on or off like sloppiness

See https://github.com/ccache/ccache/issues/1090#issuecomment-1229242525.
